core/keeper.go

```diff
@@ -134,13 +136,13 @@ func mint(evm EVMCaller, mintRequest *big.Int) error {
 	return nil
 }
 
-func triggerKeeperAndMint(evm EVMCaller, log log.Logger) {
+func triggerKeeperAndMint(evmCaller EVMCaller, log log.Logger, evm *vm.EVM) {
```

Comment:
Same thing here. You are injecting two parameters that are the same underlying value. Additionally, you are breaking the loose coupling by using a concrete type instead of the interface.